Update GPU plugin README driver section (#2105)
intel/intel-device-plugins-for-kubernetes: cmd/gpu_plugin/README.md

### KMD and UMD

There are 3 different Kernel Mode Drivers (KMDs) available:
* `i915` ([out-of-tree](https://github.com/intel-gpu/intel-gpu-i915-backports/)): official driver for Data Center GPUs, supporting only limited set of enterprise / LTS kernel versions.
* `i915` (upstream): official upstream kernel driver for older Intel client GPUs. Included in the common Linux distributions like Ubuntu.
* `xe`: official upstream kernel driver for the latest (Xe2 or newer architecture) Intel GPUs.

> Although given KMD may seem to work fine also on other hardware (HW), it's validated only for [hardware it officially supports](https://dgpu-docs.intel.com/devices/hardware-table.html), and can have problems on others. User-space APIs also differ between these KMDs, so care should taken to use in a container an User Space Driver (UMD) matching to the underlying KMD on the host.

Reviewer: the sentence does not parse well. how about:
```suggestion
Although given KMD may seem to work fine also on other hardware (HW), it's validated only for [hardware it officially supports](https://dgpu-docs.intel.com/devices/hardware-table.html), and can have problems on others. User-space APIs also differ between these KMDs, so care should be taken to ensure a container with a User Space Driver (UMD) matching to the underlying KMD on the host is used.
```

Author: Maybe:
```suggestion
Although given KMD may seem to work fine also on other hardware (HW), it's validated only for [hardware it officially supports](https://dgpu-docs.intel.com/devices/hardware-table.html), and can have problems on others. User-space APIs also differ between these KMDs, so care should taken to ensure that User Space Drivers (UMDs) within containers match KMDs used on the nodes they run, e.g. by them being installed from the same repository.
```

I'm not sure about this whole paragraph though, because user may have now way to guarantee that.  They may not even know where the UMD drivers in their (e.g. OneAPI) base containers, or KMDs on their hosts come from, so they do not know whether they match.

Nvidia way of mapping host UMD to containers avoids that issue, while causing other ones...
